prow/git/v2/remote.go

```diff
@@ -65,8 +70,11 @@ func (f *sshRemoteResolverFactory) CentralRemote(org, repo string) RemoteResolve
 
 // PublishRemote creates a remote resolver that refers to a user's remote
 // for the repository that can be published to.
-func (f *sshRemoteResolverFactory) PublishRemote(_, repo string) RemoteResolver {
-	return func() (string, error) {
+func (f *sshRemoteResolverFactory) PublishRemote(_, repo string) ForkRemoteResolver {
+	return func(forkName string) (string, error) {
+		if forkName != "" {
+			repo = forkName
```

Comment:
I believe this assignment persists, i.e. repo keeps the last non-empty forkName value passed. Think you need to create a local "r := repo" and optionally reassign that if you didn't intend for it to be persistent.

Reply:
Nice catch, that's subtle.
I fixed it, changed the variable names to distinguish which repo we're actually referring to, and added unit test coverage for this bug.